### ISSUE TITLE
Support for Equation Numbers

### ISSUE THREAD

**Issue by findmyway** [shd101wyy/vscode-markdown-preview-enhanced#67]
I use mathjax to render equations and the equation number property is missing.

![image](https://user-images.githubusercontent.com/5612003/34548561-bcd9b2c6-f13d-11e7-838a-5492b6d5cea3.png)

is rendered as:

![image](https://user-images.githubusercontent.com/5612003/34548568-c678b0a2-f13d-11e7-8f8c-d8dd7c72b32f.png)

If I open in browser, the equation is shown, but equation number is still missing

![image](https://user-images.githubusercontent.com/5612003/34548621-10119440-f13e-11e7-8d64-d0298fc43a82.png)

So, when I use `$eqref(2.4)$`, it is rendered as:

![image](https://user-images.githubusercontent.com/5612003/34548647-2a461228-f13e-11e7-8d9d-ea88f82e6015.png)

Usually, I just need to add the following script in html to add the support:

```html
<script type="text/x-mathjax-config"> MathJax.Hub.Config({ TeX: { equationNumbers: { autoNumber: "AMS" } } }); </script>
``` 

But I can't find any configuration for it. It would be great to add this configuration support.

I really appreciate your work and now I use it for my daily notes.
Thank you!

**Comment by shd101wyy** [comment]
Yes that's a bug that exists a long time ago.
You can configure MathJax by running command `Markdown Preview Enhanced: Open MathJax Config`.
However, the equation number will increase incorrectly.
I still haven't found a solution to that.
Thanks

**Comment by findmyway** [comment]
I'm not sure if the link bellow is useful:

http://docs.mathjax.org/en/latest/advanced/typeset.html#reset-equation-numbers

**Comment by ketozhang** [comment]
I was able to get it to work _(kinda)_. I rendered the following code,

## Json
```js
  TeX: {
    extensions: ['AMSmath.js', 'AMSsymbols.js', 'noErrors.js', 'noUndefined.js'],
    equationNumbers: {
      autoNumber: "AMS"
    }
```
## Markdown
```latex
The following is a blocked equation in `equation` environment.

$$
\begin{equation}
    F = ma 
    \label{eq: test}
\end{equation}
$$

Reference to Equation $\eqref{eq: test}$
```

## User Settings

If you were not to use the Pandoc Parser then it works and renders this for few seconds.

` "markdown-preview-enhanced.usePandocParser": false,`
![image](https://user-images.githubusercontent.com/13506487/34750824-18bd5a42-f55e-11e7-8b11-3e0fcd2469a4.png)

After a few seconds it reverts back to verbatim and becomes this,

![image](https://user-images.githubusercontent.com/13506487/34750878-53b16daa-f55e-11e7-9d67-1f559e9c484f.png)

However using `open in browser` it would render it properly.

It is confusing why Pandoc Parser doesn't work. To me pandoc is the strongest and closest to original LaTeX. @shd101wyy hopefully you can give us some insight.

` "markdown-preview-enhanced.usePandocParser": true,`
![image](https://user-images.githubusercontent.com/13506487/34750812-fd88d3f0-f55d-11e7-8d0c-1842aabe23bc.png)

**Comment by ketozhang** [comment]
@shd101wyy 

Any comment on why the MPE Preview is different than "Open in Browser". Browser works really well and does not require auto-reset numbering. Is there a way to use browser preview which updates as I make changes?

**Comment by shd101wyy** [comment]
@ketozhang 

Preview will only load dependencies like `MathJax.js` **once** when it is opened. The MathJax numbering is not working correctly because there is no reset.
No, there is no way to use browser preview now.

**Comment by ketozhang** [comment]
@shd101wyy 

So I'm assuming you've given up trying to fix labeling equations?
Once again the preview does render the labels correctly, however fails to keep it stable if the preview autorefresh.

**Comment by shd101wyy** [comment]
Might find a solution to that in the future, but not now. I will be busy in
the following few months.

Never give up :)

**Comment by RoyiAvital** [comment]
It doesn't work even in the case of no Auto Numbering.

For example:

```
In the [DFT Transform](https://en.wikipedia.org/wiki/Discrete_Fourier_transform) the connection between time and frequency is given by:

$$ x \left[ n \right] = \frac{1}{N} \sum_{k = 0}^{N - 1} X \left[ k \right] {e}^{j 2 \pi \frac{k}{N} n } \tag{EqnIdft} \label{EqnIdft} $$

In $ \eqref{EqnIdft} $ we can see...
```

This will render correctly inside StackExchange network yet not on MarkDownPreview Enhanced.

It seems the label creates errors regardless of numbering.

**Comment by GitAndGithub** [comment]
I have found a way that works for myself. The method is adding `$\def\label#1{}$` at the end of the markdown file. However, I don't know the reason why it works.
`This is the beginning.

$$
\begin{equation}
s=1\label{eqn1}\tag{1}\end{equation}
$$

$$
\begin{equation}
s=2\label{eqn2}\tag{2}\end{equation}
$$

eqn1 $\eqref{eqn1}$ and eqn2 $\eqref{eqn2}$

This is the end.

$\def\label#1{}$`

![image](https://user-images.githubusercontent.com/56752048/75276965-1dd7f700-5842-11ea-8d98-1afece7c92d3.png)
